### PR TITLE
Orchestrator tests

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
This PR adds missing unit tests for our orchestrator abstraction and utils.

Additionally, it fixes a bug found by those tests regarding the cache step selection.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

### CODE REVIEW

**Review by fa9r — APPROVED** [review]
Great work, not much to add 👍

File: tests/unit/conftest.py [hunk 1]
```diff
@@ -469,3 +491,43 @@ def sample_artifact_request_model() -> ArtifactRequestModel:
         project=uuid4(),
         user=uuid4(),
     )
+
+
+@pytest.fixture
+def create_step_run(
```
[the remark this answers is not in the document]
Comment (schustmi) [reply tests/unit/conftest.py:497]: I tried using sample step model, but always needed to set some specific attributes which is why I created this "flexible" fixture. Not sure if we should keep the simple one, but probably that's easier to use in other cases that just need any kind of step run?

File: src/zenml/orchestrators/cache_utils.py [hunk 2]
```diff
@@ -131,17 +132,16 @@ def get_cached_step_run(cache_key: str) -> Optional["StepRunResponseModel"]:
...
-        )
-        .items
+    client = Client()
+
+    list_method = functools.partial(
+        client.list_run_steps,
+        project_id=client.active_project.id,
+        cache_key=cache_key,
+        status=ExecutionStatus.COMPLETED,
+        sort_by="created",
```
[the remark this answers is not in the document]
Comment (schustmi) [reply src/zenml/orchestrators/cache_utils.py:142]: I'll add a todo to replace this with a descending sort once we've implemented that.

File: tests/unit/conftest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fa9r) [reply tests/unit/conftest.py:497]: Feels a bit redundant but up to you

[the remark this answers is not in the document]
Comment (schustmi) [reply tests/unit/conftest.py:473]: I just checked and the `sample_step_model` fixture was only used in a single place, so I replaced it with the new "generic" one.

**Pull request merged**